### PR TITLE
feat: Include priority fee instruction with SVM warp transfers

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

Update adapters to include priority fee instruction with sealevel collateral warp transfers

### Drive-by changes

Hoist const to top of sealevel adapter file

### Related issues

Fixes https://github.com/hyperlane-xyz/hyperlane-warp-ui-template/issues/264

### Backward compatibility

Yes

### Testing

Tested in Warp UI

### CODE REVIEW

File: typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts [hunk 1]
```diff
@@ -297,11 +312,13 @@ export abstract class SealevelHypTokenAdapter
     });
 
     const setComputeLimitInstruction = ComputeBudgetProgram.setComputeUnitLimit(
-      {
-        units: TRANSFER_REMOTE_COMPUTE_LIMIT,
-      },
+      { units: TRANSFER_REMOTE_COMPUTE_LIMIT },
     );
 
+    const setPriorityFeeInstruction = ComputeBudgetProgram.setComputeUnitPrice({
```
Comment (paulbalaji) [inline typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts:318]: nit: would add link to the phantom priority fee docs https://docs.phantom.app/developer-powertools/solana-priority-fees#how-phantom-applies-priority-fees-to-dapp-transactions

**Review by paulbalaji — APPROVED** [review]

File: typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts [hunk 2]
```diff
@@ -484,6 +502,34 @@ export abstract class SealevelHypTokenAdapter
       this.warpProgramPubKey,
     );
   }
+
+  async getMedianPriorityFee(): Promise<number | undefined> {
```
Comment (tkporter) [inline typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts:506]: nit: documenting this returns micro lamports would probably be useful for future readers

File: typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts [hunk 3]
```diff
@@ -484,6 +502,34 @@ export abstract class SealevelHypTokenAdapter
...
+    this.logger.debug('Fetching priority fee history for token transfer');
+
+    const collateralAddress = this.addresses.token;
+    const fees = await this.getProvider().getRecentPrioritizationFees({
```
Comment (tkporter) [inline typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts:510]: Realizing now that I guess Jito transactions wouldn't show up in this bc they circumvent the typical prioritization fee payment method. This is totally fine, just calling it out

File: typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts [hunk 4]
```diff
@@ -51,6 +52,28 @@ import {
...
+ * The factor by which to multiply the median prioritization fee
+ * instruction added to transfer transactions.
+ */
+const PRIORITY_FEE_PADDING_FACTOR = 2;
+
+/**
+ * The minimum priority fee to use if the median fee is
+ * unavailable or too low.
+ */
+const MINIMUM_PRIORITY_FEE = 10_000;
```
Comment (tkporter) [inline typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts:75]: for anyone else curious, this results in a 0.000001 SOL priority fee

**Review by tkporter — APPROVED** [review]

**Pull request merged**